src/spartan_app.erl

```diff
@@ -32,35 +29,48 @@ start(_StartType, _StartArgs) ->
     maybe_start_tcp_listener(),
     Ret.
 
-%%--------------------------------------------------------------------
 stop(_State) ->
     ranch:stop_listener(?TCP_LISTENER_NAME),
     ok.
 
+%% @doc Wait for a response.
+wait_for_reqid(ReqID, Timeout) ->
+    receive
+        {ReqID, ok} ->
+            ok;
+        {ReqID, ok, Val} ->
+            {ok, Val}
+    after Timeout ->
+        {error, timeout}
+    end.
 
-%% @doc Parse an IP Address
+%%====================================================================
+%% API
+%%====================================================================
+
+%% @doc Parse an IPv4 Address
 parse_ipv4_address(Value) when is_binary(Value) ->
     parse_ipv4_address(binary_to_list(Value));
 parse_ipv4_address(Value) ->
     {ok, IP} = inet:parse_ipv4_address(Value),
     IP.
 
+%% @doc Parse an IPv4 Address with an optionally specified port.
+%% The default port will be substituted in if not given.
+parse_ipv4_address_with_port(Value, DefaultPort) ->
```

Comment:
Good idea. I'll type spec all of the parse_* functions for good measure since dialyzer is run by the rebar3 build.